Mathlib/Order/RelClasses.lean

```diff
@@ -244,6 +244,30 @@ lemma WellFounded.prod_lex {ra : α → α → Prop} {rb : β → β → Prop} (
     (hb : WellFounded rb) : WellFounded (Prod.Lex ra rb) :=
   (Prod.lex ⟨_, ha⟩ ⟨_, hb⟩).wf
 
+section PSigma
+
+open PSigma
+
+/-- The lexicographical order of well-founded relations is well-founded. -/
+theorem WellFounded.psigma_lex
+    {α : Sort*} {β : α → Sort*} {r : α → α → Prop} {s : ∀ a : α, β a → β a → Prop}
+    (ha : WellFounded r) (hb : ∀ x, WellFounded (s x)) : WellFounded (Lex r s) :=
+  WellFounded.intro fun ⟨a, b⟩ => lexAccessible (WellFounded.apply ha a) hb b
+
+theorem WellFounded.psigma_revLex
```

Comment:
This also was phrased for Sort*, can you re-add this (or say why this is not desirable), please?